### PR TITLE
Add Credential Revocation Contracts 

### PR DESCRIPTION
**Pull request by vdsilveira**
**Summary of Changes:**

***Addition of Contracts for Credential Revocation:***

- Added a directory named revoke to the contracts directory, this directory contains the smart contracts related to credential revocation.
 - Included the revocation contract in genesis.json.

***Modifications to Other Contracts:***

- Added errors related to revocation in AnouncredsErrors.sol.

 - Modified the ecrecover method to use the standardized ECDSA method from the OpenZeppelin library in the following contracts:
   - CredentialDefinitionRegistry
   - CredentialDefinitionRegistryInterface
   - SchemaRegistry
   - SchemaRegistryInterface
   - IndyDidRegistry
   - IndyDidRegistryInterface

### CODE REVIEW

File: smart_contracts/contracts/did/IndyDidRegistry.sol [hunk 1]
```diff
@@ -2,57 +2,37 @@
...
 
-    /**
-     * @dev Mapping DID to its corresponding DidRecord (Document/Metadata).
-     */
-    mapping(address identity => DidRecord didRecord) private _dids;
+    mapping(address => DidRecord) private _dids;
 
-    /**
-     * Checks that DID already exists
-     */
```
Comment (Toktar) [inline smart_contracts/contracts/did/IndyDidRegistry.sol:23]: Could you not to remove the codebase documentation?

File: smart_contracts/contracts/did/IndyDidRegistry.sol [hunk 2]
```diff
@@ -75,51 +52,55 @@ contract IndyDidRegistry is IndyDidRegistryInterface, ControlledUpgradeable {
...
 
-    /// @inheritdoc IndyDidRegistryInterface
-    function resolveDid(address identity) public view virtual _didExist(identity) returns (DidRecord memory didRecord) {
+    function resolveDid(address identity) public view _didExist(identity) returns (DidRecord memory didRecord) {
         return _dids[identity];
     }
 
     function _createDid(
         address identity,
-        address actor, // actor is either message sender in case of `createDid` or signer in case of `createDidSigner`
```
Comment (Toktar) [inline smart_contracts/contracts/did/IndyDidRegistry.sol:122]: The same ask for comments and documentation

**Review by Toktar — CHANGES_REQUESTED** [review]
@vdsilveira Thank you a lot! We were thinking about adding revocations for a long time.
Right now, this implementation looks like DID Document status that already exists. Also, I have some questions about using credential identifiers.
Could you please start with design and when we discuss it, add tests, SDK functionality, wrappers and examples/demo?
Glad to have the opportunity to work with you

**Comment by vdsilveira** [comment]
I'm closing this PR to refactor the code, including the tests, wrappers and so on.

**Pull request closed without merging**